### ISSUE TITLE
Plaits Firmware Update released

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#7124]
Over at https://github.com/pichenettes/eurorack/blob/master/plaits/dsp/voice.cc you can see that mutable has released the firmware for the expanded plaits modes. We can and should integrate this with surge. A few tricks, though

1. Gotta update that submodule fork to the new code. Easy
2. Patch compatibility is mildly tricky since the engine orders change. So up the streaming version when we make this change and go ahead and do an unstream handler
3. Rack is also a bit tricky since it needs a separate unstream handler, so remember to do that
4. Fix the engine UI things
5. Most importantly, you can have user wavetables and FM models. Think about if or how we support that. (In theory twist could support a 'wavetable' which is the sysex for the models. but requires thought.

Dont know if we do this for 1.3, so putting here in 1.3 for now so I can stare at it and remove it from the milestone towards the end, to be honest.

**Comment by mkruselj** [comment]
I'm not convinced we should allow the whole DX7 mode tho...

**Comment by baconpaul** [comment]
Yeah I sort of agree. There’s other new modes. But anyway just a placeholder issue in case folks ask since I saw the software was now on GitHub.

**Comment by mkruselj** [comment]
OK here's my thoughts:

1 Classic waveshapes with filter **YES**
2 Phase distortion and modulation **YES**
3...5 6-OP FM **NO**
6 Wave Terrain **YES**
7 String Machine **YES**
8 4 squares for chords or arps **YES** although there's a specificity that Timbre attenuverter acts as an additional parameter to change envelope shapes. Probably figure out which one is the most neutral that acts as a basically always on gate, if it even exists. If it doesn't, it's a **NO**.

**Comment by baconpaul** [comment]
Few notes

1: We have a fork of the euro rack repo already which basically strips out most of the stuff to make clones faster. So we need to update plaits there
2. Since this is a fork we can pick and choose the models to add, per the above
3. once we do we want to update the streaming revision ff_revision and then do a handleStreamingMismatches in TwistOscillator.cpp but lets *also* add a define somewhere like `TWIST_MODULE_VERSION` which we make `20230803` or something so I can hack rack somehow and live when we upgrade.

**Comment by baconpaul** [comment]
As predicted, did bounce out of milestone.
